examples/form/src/CreateOrder.tsx

```diff
@@ -0,0 +1,52 @@
+import { OrderForm } from "./OrderForm.tsx";
+import { useAccount } from "./main.tsx";
+import {
+  BubbleTeaAddOnTypes,
+  BubbleTeaBaseTeaTypes,
+  BubbleTeaOrder,
+  ListOfBubbleTeaAddOns,
+} from "./schema.ts";
+
+export interface BubbleTeaOrderType {
+  baseTea: (typeof BubbleTeaBaseTeaTypes)[number];
+  addOns: Array<(typeof BubbleTeaAddOnTypes)[number]>;
+  deliveryDate: Date;
+  withMilk: boolean;
+  instructions?: string;
+}
+
+export function CreateOrder() {
+  const { me } = useAccount({
+    profile: { orders: [] },
+  });
+
+  if (!me?.profile) return;
+
+  const newOrder = BubbleTeaOrder.create(
+    {
+      baseTea: BubbleTeaBaseTeaTypes[0],
+      addOns: ListOfBubbleTeaAddOns.create([], { owner: me.profile._owner }),
+      deliveryDate: new Date(),
+      withMilk: false,
+    },
+    { owner: me?.profile._owner },
+  );
```

Comment:
It's not great to call create in the component body, this way a new order gets created on each render.

The React-friendly way to do this is:
```ts
const [newOrder, setNewOrder] = useState<BubbleTeaOrder | undefined>();

const publicGroup = me?.profile._owner;
useEffect(() => {
  if (!publicGroup) return;
  
   const newOrder = BubbleTeaOrder.create({
      baseTea: BubbleTeaBaseTeaTypes[0],
      addOns: ListOfBubbleTeaAddOns.create([], { owner: me.profile._owner }),
      deliveryDate: new Date(),
      withMilk: false,
    },
    { owner: publicGroup },
  );
  
  setNewOrder(newOrder);
}, [publicGroup])

if (!newOrder) return null;
```

Reply:
@aeplay This could become something we provide as hook:

```ts
const newOrder = useCoStateCreate((me) => {
   const publicGroup = me.profile._owner;
   
   return BubbleTeaOrder.create({
      baseTea: BubbleTeaBaseTeaTypes[0],
      addOns: ListOfBubbleTeaAddOns.create([], { owner: me.profile._owner }),
      deliveryDate: new Date(),
      withMilk: false,
    },
    { owner: publicGroup }
    );
})
```

This way we remove the ugly waiting for the account loading.

Reply:
Good idea to introduce a hook that helps with the state handling.

In general, the hard part with the create + edit form is when you have fields that are required on a created CoValue, but obviously need to be optional _while creating the CoValue_. I think in this example we conveniently avoid this, but it would be good to introduce one required field that doesn't have an obvious "empty" state, so we are forced to find a solution to and can offer a helpful pattern @trishalim @gdorsi.